packages/interchain/interchain-core/src/packet.rs

```diff
@@ -0,0 +1,335 @@
+                    .iter()
+                    .find_map(|packet_result| match &packet_result {
+                        IbcPacketOutcome::Timeout { timeout_tx } => {
+                            timeout_tx.event_attr_value(event_type, attr_key).ok()
+                        }
+                        IbcPacketOutcome::Success {
+                            receive_tx,
+                            ack_tx,
+                            ack: _,
+                        } => receive_tx
+                            .event_attr_value(event_type, attr_key)
+                            .or_else(|_| ack_tx.event_attr_value(event_type, attr_key))
+                            .ok(),
+                    })
+                    .ok_or(StdError::generic_err(format!(
+                        "event of type {event_type} does not have a value at key {attr_key}"
+                    )))
+            })
+    }
+
+    fn event_attr_values(&self, event_type: &str, attr_key: &str) -> Vec<String> {
+        let mut all_results = self.tx_id.response.event_attr_values(event_type, attr_key);
+
+        all_results.extend(self.packets.iter().flat_map(|packet_result| {
+            match &packet_result {
+                IbcPacketOutcome::Timeout { timeout_tx } => {
+                    timeout_tx.event_attr_values(event_type, attr_key)
+                }
+                IbcPacketOutcome::Success {
+                    receive_tx,
+                    ack_tx,
+                    ack: _,
+                } => [
+                    receive_tx.event_attr_values(event_type, attr_key),
+                    ack_tx.event_attr_values(event_type, attr_key),
+                ]
+                .concat(),
+            }
+        }));
+
+        all_results
+    }
+
+    fn data(&self) -> Option<Binary> {
+        unimplemented!("No data fields on Ibc Tx Analysis")
+    }
+}
+
+pub mod success {
+    use crate::{ack_parser::polytone_callback::Callback, tx::TxId};
+    use cosmwasm_std::Empty;
+    use cw_orch_core::environment::CwEnv;
+
+    #[derive(Debug, PartialEq, Clone)]
+    pub enum SuccessfullAck<CustomOutcome = Empty> {
+        Polytone(Callback),
+        Ics20,
+        Ics004(Vec<u8>),
+        Custom(CustomOutcome),
+    }
+
+    impl SuccessfullAck<Empty> {
+        pub fn into_custom<CustomOutcome>(self) -> SuccessfullAck<CustomOutcome> {
+            match self {
+                SuccessfullAck::Polytone(callback) => SuccessfullAck::Polytone(callback),
+                SuccessfullAck::Ics20 => SuccessfullAck::Ics20,
+                SuccessfullAck::Ics004(vec) => SuccessfullAck::Ics004(vec),
+                SuccessfullAck::Custom(_) => unreachable!(),
+            }
+        }
+    }
+
+    /// Success packet outcome. This is the result of a packet analysis.
+    /// The T generic is used to allow for raw transactions or analyzed transactions to be used
+    #[derive(Debug, PartialEq, Clone)]
+    pub struct SuccessIbcPacketOutcome<T, CustomOutcome = Empty> {
```

Comment:
I'm a bit confused on what is meant with "success". Does it mean that the ibc packet got acknowledged? On the topic of naming I think we should also use `Result` over `Outcome`, or maybe `Status` as it's only really a result after it's asserted.

What this datatype wants to reflect is a packet that has succesfully gone through its lifecycle. So mabye `AcknowledgedIbcPacket` would be even better?

Reply:
Success is not very well defined for now, but it should be very well defined on this crate. For now, usage has shown that success necessitates at least : 
- Packet has been acknowledged
- Acknowledgement has been successfully parsed and the parsed ack corresponds to a successful IBC operation

We could also add : 
- Acknowledgment execution has been successful (though it's way more difficult to know because ack transactions can't fail). 

So it's not a result (which in Rust semantics has 2 possible outcomes), it's an Result::Ok, if you will, so a Success here. 
- AcknowledgedIbcPacket is a little limited considering the above
- Status is also limited

Reply:
> Acknowledgement has been successfully parsed and the parsed ack corresponds to a successful IBC operation

Does it make sense to have a separate step between `RawAck` and `AssertedAck`? 

I.e. why would we want a type that we know is a packet that went through its cycle and is of a specific datastructure but that we don't know the result of? 

Can we omit the type and simply talk about acknowledgements that were committed (no guarantees on the formatting or result of the packet) and then a checked type that does the whole parsing and checking into a result?